File: java/sdk/src/main/java/com/dapr/sdk/Controllers/PublishController.java
package com.dapr.sdk.Controllers;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.UUID;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import io.dapr.client.DaprClient;
import io.dapr.client.DaprClientBuilder;

@RestController
public class PublishController {
    @GetMapping("/api/pub")
    public ResponseEntity<String> publish() {
        UUID uuid = UUID.randomUUID();
        Map<String, String> orders = new HashMap<>();
        orders.put("orderId", uuid.toString());

        List<Map<String, String>> data = new ArrayList<>();
        data.add(orders);

        DaprClient client = new DaprClientBuilder().build();
        client.publishEvent(
                "pubsub",
                "orders",
                orders).block();

        return new ResponseEntity<String>("Order published with order id: " + uuid.toString(), HttpStatus.OK);
    }
}

File: php/sdk/app/Http/Controllers/PublishController.php
<?php

namespace App\Http\Controllers;

use Illuminate\Support\Str;

class PublishController extends Controller
{
    public function publish()
    {
        $id = (string) Str::uuid();
        $arr = array('orderId' => $id);
        $data = [$arr];

        $daprClient = \Dapr\Client\DaprClient::clientBuilder()->build();
        $daprClient->publishEvent('pubsub', 'orders', $data);

        return json_encode('Published order with id: ' . $id);
    }
}
